runtime/near-evm-runner/src/types.rs

```diff
@@ -96,6 +102,124 @@ pub struct MetaCallArgs {
     pub input: Vec<u8>,
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub struct EthTransaction {
+    pub nonce: U256,
+    pub gas_price: U256,
+    pub gas: U256,
+    pub to: Option<Address>,
+    pub value: U256,
+    pub data: Vec<u8>,
+}
+
+impl EthTransaction {
+    pub fn rlp_append_unsigned(&self, s: &mut RlpStream, chain_id: Option<u64>) {
+        s.begin_list(if chain_id.is_none() { 6 } else { 9 });
+        s.append(&self.nonce);
+        s.append(&self.gas_price);
+        s.append(&self.gas);
+        match self.to.as_ref() {
+            None => s.append(&""),
+            Some(address) => s.append(address),
+        };
+        s.append(&self.value);
+        s.append(&self.data);
+        if let Some(chain_id) = chain_id {
+            s.append(&chain_id);
+            s.append(&0u8);
+            s.append(&0u8);
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct EthSignedTransaction {
+    pub transaction: EthTransaction,
+    pub v: u64,
+    pub r: U256,
+    pub s: U256,
+}
```

Comment:
Add comments here

Reply:
Resolved in 311b3c8ce1c18d1662e5549575f2b3ad8657a21a.